pkg/auth/iam/iam.go

```diff
@@ -265,6 +277,9 @@ func (filter *Filter) validateRefererHeader(request *restful.Request, claims *ia
 	refererHeader := request.HeaderParameter(refererHeaderKey)
 	clientRedirectURIs := strings.Split(clientInfo.RedirectURI, ",")
 	for _, redirectURI := range clientRedirectURIs {
+		if !filter.options.StrictRefererHeaderValidation {
```

Comment:
correct, will handle it

Reply:
why the option will be nil?

Reply:
It will possible if the client / service initializes the filter with nil options, we should handle it to avoid the error

Reply:
from my side, this is not a important issue, because if the integrator set the value to nil, the whole service will panic in all the endpoints, developer has not correct the value. 

for enhancement, we can add a nil check in the init method to prevent this happen

Reply:
that's why it will important, just imagine this problem happens in production especially in cloud team, afaik, we cannot easily rollout the service there
and we cannot catch the issue during the deployment because the panic will happen if client calls the endpoint
yeah, we need to add check in the init

Reply:
do you mean after developer init the filter and will not do any test?

Reply:
this is a obvious problem that the developer not correct set the value, it's the error, not bug.